sm64-tas-scripting/Inputs.cpp

```diff
@@ -329,57 +361,48 @@ int M64::load()
 	return 1;
 }
 
-bool Inputs::HauEquals(int16_t angle1, int16_t angle2)
-{
-	int16_t hau1 = angle1 - (angle1 & 15);
-	int16_t hau2 = angle2 - (angle2 & 15);
 
-	return hau1 == hau2;
-}
 
 int M64::save(long initFrame)
 {
-	FILE* f;
+	std::ofstream f(fileName, ios_base::binary);
+	f.exceptions(ios_base::failbit | ios_base::badbit);
 
 	uint16_t buttons;
 	int8_t stick_x, stick_y;
 
 	size_t err;
+	if (frames.empty())
+		return 1;
```

Comment:
```suggestion
	if (frames.empty())
		return 1;
	std::ofstream f(fileName, ios_base::binary);
	f.exceptions(ios_base::failbit | ios_base::badbit);

	uint16_t buttons;
	int8_t stick_x, stick_y;

	size_t err;

```

don't open the file if you aren't going to write it, else the file is truncated to nothing